### ISSUE TITLE
Support question: Escaping for sqlite FTS5

### ISSUE THREAD

**Issue by alinsavix** [coleifer/peewee#2955]
Hi! Lovely package you've got here, it's made working with databases in python actually fun again!

I've just run into one issue that I've not managed to solve: How the heck does one correctly escape quote marks for FTS5 MATCH queries? Normally I'd expect peewee to do all the escaping under the hood for me, but that seems to not be happening in this case, and nothing I can think of to do myself actually improves on the situation.

Minimal test case:
```
#! /usr/bin/env python
from playhouse.sqlite_ext import (FTS5Model, RowIDField, SearchField, SqliteExtDatabase)
db = SqliteExtDatabase("test.db")

class TestIndex(FTS5Model):
    rowid = RowIDField()
    title = SearchField()

    class Meta:
        database = db

db.create_tables([TestIndex])

q = TestIndex.select().where(TestIndex.match("who's on first"))
print(q[0])
```

When run, this produces:

```
  File "D:\<path>\venv\Lib\site-packages\peewee.py", line 3322, in execute_sql
    cursor.execute(sql, params or ())
sqlite3.OperationalError: fts5: syntax error near "'"
```
I've tried escaping the quote mark by doubling it, by backslash prefixing it, by double-backslash prefixing it, etc, but not actually found anything that works.

What am I missing?

On the same topic, it really feels like FTS5 is absolutely full of landmines when handling user input (e.g. if the user includes a `-` in their search string, things behave differently), are you aware of any good way to escape/strip/sanitize a user-provided string to make it behave predictably with FTS5, without requiring the end user to know that FTS5 even exists? "Search for user-provided input with FTS5" must be an extremely common use case, so surely there must be *some* known way to make those searches behave predictably... right? I sure couldn't find anything, though, so figured I'd ask...

**Comment by coleifer** [comment]
>On the same topic, it really feels like FTS5 is absolutely full of landmines when handling user input

Agreed, that's been my experience as well - I suggest carefully reading and reviewing https://sqlite.org/fts5.html#full_text_query_syntax and implementing your own sanitizing logic on top of it.

We provide a simple helper to clean a query, see `FTS5Model.clean_query()`, which is used in turn by the `FTS5Model.search()` method.

Unfortunately, FTS5 is a lot more strict, and it is left to the user to handle tokenizing and cleaning in a sensible way. Feel free to reach out to the sqlite mailing list, though, they may have more insight into how to handle this: https://sqlite.org/forum/forum

**Comment by alinsavix** [comment]
Aha, FTS5Model.clean_query() actually takes care of like 95% of what I care about, awesome. Thanks!
